### PR TITLE
Fix!(tests): normalize identifiers according to the config's dialect

### LINKED ISSUE #1205: Issue with `snowflake`  dialect when following the "Get started" instructions
Hello,

When I follow the instructions of the "Get started" section in the docs, I have the following error when running `sqlmesh plan`:

```
$ sqlmesh plan
Column '"ID"' could not be resolved for '/Users/soheil.salmani/Desktop/Activities/sqlmesh-activities/sqlmesh-example/models/incremental_model.sql', the column may not exist or is ambiguous
Error: Model 'sqlmesh_example.full_model' was not found at /Users/soheil.salmani/Desktop/Activities/sqlmesh-activities/sqlmesh-example/tests/test_full_model.yaml
```

I believe it's a problem with the `snowflake` dialect, because when I change to `bigquery` (in  the `config.yaml` file), it works:

```
$ sqlmesh plan
======================================================================
Successfully Ran 1 tests against duckdb
----------------------------------------------------------------------
New environment `prod` will be created from `prod`
Summary of differences against `prod`:
└── Added Models:
    ├── sqlmesh_example.full_model
    ├── sqlmesh_example.incremental_model
    └── sqlmesh_example.seed_model
Models needing backfill (missing dates):
├── sqlmesh_example.seed_model: 2023-07-22 - 2023-07-22
├── sqlmesh_example.full_model: 2020-01-01 - 2023-07-22
└── sqlmesh_example.incremental_model: 2020-01-01 - 2023-07-22
Apply - Backfill Tables [y/n]: ^C
Aborted!
```

### PR DESCRIPTION
**Pull request by GeorgeSittas**
Fixes #1205

- [x] Normalize various needed model meta fields.
- [x] Normalize unit test identifiers to fix resolution discrepancies.
- [x] Add tests.

### CODE REVIEW

File: sqlmesh/core/model/seed.py [hunk 1]
```diff
@@ -49,9 +51,16 @@ def _get_df(self) -> pd.DataFrame:
                 index_col=False,
                 on_bad_lines="error",
             )
+            self._df = self._df.rename(
+                columns={
+                    col: normalize_identifiers(col, dialect=self.dialect).name
+                    for col in list(self._df.columns)
```
Comment (tobymao) [inline sqlmesh/core/model/seed.py:57]: what's this list call?

Comment (GeorgeSittas) [reply sqlmesh/core/model/seed.py:57]: oversight, will get rid of it

**Review by tobymao — APPROVED** [review]

File: sqlmesh/core/dialect.py [hunk 2]
```diff
@@ -701,7 +703,16 @@ def pandas_to_sql(
...
+    )
+
+
+def normalize_identifiers(
+    value: str | exp.Expression, dialect: DialectType = None
+) -> exp.Expression:
+    return normalize_identifiers_func(
+        value if isinstance(value, exp.Expression) else exp.to_identifier(value),
+        dialect=dialect,
     )
```
Comment (GeorgeSittas) [inline sqlmesh/core/dialect.py:716]: Do you think this is worth moving over to SQLGlot? The only difference with sqlglot's `normalize_identifiers` is that it knows how to handle strings, which was useful for validating the model's meta fields in this case.

Comment (tobymao) [reply sqlmesh/core/dialect.py:716]: sure

**Review by tobymao — APPROVED** [review]

**Pull request merged**